src/pages/Login.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react'
+import ExternalHeader from '../components/ExternalHeader'
+import Footer from '../components/Footer'
+
+import '../App.sass';
+
+const fetch = require("node-fetch");
+const URLSearchParams = require("url-search-params");
+const Redirect = require("react-router-dom").Redirect;
+
+class Login extends React.Component {
+    constructor(props){
+      super(props)
+      this.state = {
+        newAccount: true,
+        name: null,
+        email: null,
+        password: null,
+        accountId: null
+      }
+
+      this.updateInput = this.updateInput.bind(this)
+      this.onSubmit = this.onSubmit.bind(this)
+      this.swapMode = this.swapMode.bind(this)
+    }
+
+    updateInput(event) {
+        const name = event.target.name
+        this.setState({[name] : event.target.value})
+    }
+
+    onSubmit() {
+        if (!this.state.name && this.state.newAccount) {
+            alert("Name is required.")
+            return
+        } else if (!this.state.email) {
+            alert("Email is required.")
+            return
+        } else if (!this.state.password) {
+            alert("Password is required.")
+            return
+        }
+
+        var formBody = new URLSearchParams();
+        if (this.state.newAccount) {
+            formBody.append('name', this.state.name);
+        }
+        formBody.append('email', this.state.email);
+        formBody.append('password', this.state.password);
+
+        var url = (this.state.newAccount ? "https://api.pennlabs.org/portal/account/new" : "https://api.pennlabs.org/portal/account/login")
+        fetch((url), {
+            method: 'POST',
+            headers: {
+                'Accept': 'application/json',
+                'Content-Type': 'application/x-www-form-urlencoded'
+            },
+            body: formBody
+        })
+        .then((response) => response.json())
+        .then((json) => {
+            if (json.error) {
+                alert(json.error)
+            } else if (json.msg) {
+                alert(json.msg)
+            } else if (json.account_id) {
+                this.setState({accountId: json.account_id})
+                window.sessionStorage.setItem('accountID', json.account_id)
```

Comment:
Instead of session storage, use cookies with an expiration date. I'd suggest the[ js-cookie library](https://www.npmjs.com/package/js-cookie), which has a nice API. The reason we don't want to use sessionStorage is that it only lasts as long as the browser is open. If the user closes the browser, they will have to log in again.